sycl/include/CL/sycl.hpp

```diff
@@ -11,6 +11,8 @@
 #include <CL/sycl/accessor.hpp>
 #include <CL/sycl/atomic.hpp>
 #include <CL/sycl/backend.hpp>
+#include <CL/sycl/backend/cuda.hpp>
+#include <CL/sycl/backend/opencl.hpp>
```

Comment:
Wouldn't it be better to include those 2 backends (cuda and opencl) inside backend.hpp?
Otherwise, looks good to me, I don't have any other comments.

Reply:
Can't really agree with it as we will get some compilation error: `struct interop` template isn't yet available. The combined source will look like this:
```c++
// backend.hpp start

// backend/opencl.hpp start
// backend.hpp isn't really included now, 'cause due to pragma once
template <>
struct interop<backend::opencl, cl::sycl::queue> {
  using type = ...;
};
...
// backend/opencl.hpp end

...

template <backend Backend, typename SYCLType> struct interop;

// backend.hpp end
```

Though, one may want to include `opencl.hpp` and `cuda.hpp` at the end of `backend.hpp`. It looks like some bad style for me and I bet you'll get some compilation error, then.